fe/fe-core/src/main/java/com/starrocks/http/rest/RestBaseAction.java

```diff
@@ -93,6 +99,25 @@ public void handleRequest(BaseRequest request) {
         }
     }
 
+    @VisibleForTesting
+    public String getErrorRespWhenUnauthorized(AccessDeniedException accessDeniedException) {
+        if (Strings.isNullOrEmpty(accessDeniedException.getMessage())) {
+            ConnectContext context = ConnectContext.get();
+            if (context != null) {
+                AuthorizationMgr authorizationMgr = GlobalStateMgr.getCurrentState().getAuthorizationMgr();
+                UserIdentity userIdentity = context.getCurrentUserIdentity();
+                List<String> activatedRoles = authorizationMgr.getRoleNamesByRoleIds(context.getCurrentRoleIds());
+                List<String> inactivatedRoles =
+                        authorizationMgr.getInactivatedRoleNamesByUser(userIdentity, activatedRoles);
+                return "Access denied for user " + userIdentity  + ". " +
+                        String.format(ErrorCode.ERR_ACCESS_DENIED_HINT_MSG_FORMAT, activatedRoles, inactivatedRoles);
+            }
+            return "Access denied.";
+        } else {
```

Comment:
looks like this short `else` branch can be short cut at the very beginning, 

```
if (!Strings.isNullOrEmpty(...)) {
    return ...
}
// continue processing when the exception doesn't have error message.

Reply:
make sense